File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        unordered_set<int> st;
        int ans = 0;
        int currSum = 0;
        int x = 1;
        for(int &i: banned)
            st.insert(i);
        while(x <= n && currSum <= maxSum){
            if(st.find(x) == st.end()){
                ans++;
                currSum += x;
            }
            x++;
        }
        if(currSum > maxSum) return ans-1;
        return ans;
    }
};

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        st = set()
        ans = 0
        currSum = 0
        x = 1
        m = len(banned)
        for i in range(m):
            st.add(banned[i])
        while x <= n and currSum <= maxSum:
            if x not in st:
                currSum += x
                ans +=1
            x+=1
        if(currSum > maxSum):
            ans-= 1
        return ans
        